rotkehlchen/tests/db/test_db.py

```diff
@@ -487,33 +489,34 @@ def test_settings_entry_types(database):
             date_display_format='%d/%m/%Y %H:%M:%S %z',
             submit_usage_analytics=False,
         ))
-        res = database.get_settings(cursor)
-
-    assert isinstance(res.version, int)
-    assert res.version == ROTKEHLCHEN_DB_VERSION
-    assert isinstance(res.last_write_ts, int)
-    assert isinstance(res.premium_should_sync, bool)
-    assert res.premium_should_sync is True
-    assert isinstance(res.include_crypto2crypto, bool)
-    assert res.include_crypto2crypto is True
-    assert isinstance(res.ui_floating_precision, int)
-    assert res.ui_floating_precision == 1
-    assert isinstance(res.taxfree_after_period, int)
-    assert res.taxfree_after_period == 1
-    assert isinstance(res.balance_save_frequency, int)
-    assert res.balance_save_frequency == 24
-    assert isinstance(res.last_balance_save, int)
-    assert res.last_balance_save == 0
-    assert isinstance(res.main_currency, Asset)
-    assert res.main_currency == DEFAULT_TESTS_MAIN_CURRENCY
-    assert isinstance(res.date_display_format, str)
-    assert res.date_display_format == '%d/%m/%Y %H:%M:%S %z'
-    assert isinstance(res.submit_usage_analytics, bool)
-    assert res.submit_usage_analytics is False
-    assert isinstance(res.active_modules, tuple)
-    assert res.active_modules == DEFAULT_ACTIVE_MODULES
-    assert isinstance(res.frontend_settings, str)
-    assert res.frontend_settings == ''
+        settings = database.get_settings(cursor)
+        cache = database.get_cache(cursor)
+
+    assert isinstance(settings.version, int)
+    assert settings.version == ROTKEHLCHEN_DB_VERSION
+    assert isinstance(cache.last_write_ts, int)
+    assert isinstance(settings.premium_should_sync, bool)
+    assert settings.premium_should_sync is True
+    assert isinstance(settings.include_crypto2crypto, bool)
+    assert settings.include_crypto2crypto is True
+    assert isinstance(settings.ui_floating_precision, int)
+    assert settings.ui_floating_precision == 1
+    assert isinstance(settings.taxfree_after_period, int)
+    assert settings.taxfree_after_period == 1
+    assert isinstance(settings.balance_save_frequency, int)
+    assert settings.balance_save_frequency == 24
+    assert isinstance(cache.last_balance_save, int)
+    assert cache.last_balance_save == 0
```

Comment:
there is only this one If I'm not wrong. Maybe create a different function for this one and test the other fields?